enterprise/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/Profile.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return profileFile.hashCode();
+    }
+
+    @Override
+    public String toString()
+    {
+        return "Profile(" + profileSequenceId + " for " + pagedFile + ")";
+    }
+
+    File file()
+    {
+        return profileFile;
+    }
+
+    void delete( FileSystemAbstraction fs )
+    {
+        fs.deleteFile( profileFile );
+    }
+
+    InputStream read( FileSystemAbstraction fs ) throws IOException
+    {
+        InputStream source = fs.openAsInputStream( profileFile );
+        try
+        {
+            return new GZIPInputStream( source );
+        }
+        catch ( IOException e )
+        {
+            IOUtils.closeAllSilently( source );
+            throw new IOException( "Exception when building decompressor.", e );
+        }
+    }
+
+    OutputStream write( FileSystemAbstraction fs ) throws IOException
+    {
+        fs.mkdirs( profileFile.getParentFile() ); // Create PROFILE_FOLDER if it does not exist.
+        OutputStream sink = fs.openAsOutputStream( profileFile, false );
+        try
+        {
+            return new GZIPOutputStream( sink );
+        }
+        catch ( IOException e )
+        {
+            IOUtils.closeAllSilently( sink );
+            throw new IOException( "Exception when building compressor.", e );
+        }
+    }
+
+    Profile next()
+    {
+        long next = profileSequenceId + 1L;
+        return new Profile( profileName( pagedFile, next ), pagedFile, next );
+    }
+
+    static Profile first( File file )
+    {
+        return new Profile( profileName( file, 0 ), file, 0 );
+    }
+
+    private static File profileName( File file, long count )
+    {
+        String name = file.getName();
+        File dir = new File( file.getParentFile(), PROFILE_DIR );
+        return new File( dir, name + "." + Long.toHexString( count ) + SUFFIX_CACHEPROF );
+    }
+
+    static Predicate<Profile> relevantTo( PagedFile pagedFile )
+    {
+        return p -> p.pagedFile.equals( pagedFile.file() );
+    }
+
+    static Stream<Profile> findProfilesInDirectory( FileSystemAbstraction fs, File dir )
+    {
+        File[] files = fs.listFiles( new File( dir, PROFILE_DIR ) );
+        if ( files == null )
+        {
+            return Stream.empty();
+        }
+        return Stream.of( files ).flatMap( Profile::parseProfileName );
+    }
+
+    private static Stream<Profile> parseProfileName( File profile )
+    {
+        File profileFolder = profile.getParentFile();
+        File dir = profileFolder.getParentFile();
+        String name = profile.getName();
+        if ( !name.endsWith( SUFFIX_CACHEPROF ) )
+        {
+            return Stream.empty();
+        }
+        int lastDot = name.lastIndexOf( '.' );
+        int secondLastDot = name.lastIndexOf( '.', lastDot - 1 );
+        String countStr = name.substring( secondLastDot + 1, lastDot );
+        try
+        {
+            long sequenceId = Long.parseLong( countStr, 16 );
```

Comment:
Still hex?! At best you save 3 chars...
```
9223372036854775807 19 chars
7FFFFFFFFFFFFFFF    16 chars
```
Human readability is more important than those 3.

Reply:
It's not supposed to be human readable.

Reply:
I'm a human(last time I checked) and if I need to debug this I would prefer if the number inside the code was the same as the number in the file name. Why make our lives harder than they need to be?

Reply:
The same should hold true for all the rotated files, e.g. transaction logs. Trying to predict human behavior is bound to fail anyway so let's just make the simplest possible.